Formula/apm-server.rb

```diff
@@ -0,0 +1,99 @@
+class ApmServer < Formula
+  include Language::Python::Virtualenv
```

Comment:
virtualenv isn't needed for installing a build time resource. just use `setup_install_args`

Reply:
`setup_install_args` appears to invoke a `setup.py` and that's not what is going on here?

Reply:
I meant like this:
```

Reply:
But actually it looks like you could get away with this:
```
diff --git a/Formula/auditbeat.rb b/Formula/auditbeat.rb
index 6ff7c7e..42a9b29 100644
--- a/Formula/auditbeat.rb
+++ b/Formula/auditbeat.rb
@@ -14,13 +14,11 @@ class Auditbeat < Formula
     ENV["GOPATH"] = buildpath
     (buildpath/"src/github.com/elastic/beats").install buildpath.children
 
-    venv = virtualenv_create(buildpath)
-    venv.pip_install "virtualenv"
-    ENV["PATH"] = ENV["PATH"].split(":").push(buildpath/"bin").join(":")
+    venv = virtualenv_create(buildpath/"venv")
 
     cd "src/github.com/elastic/beats/auditbeat" do
       system "make"
-      system "make", "update"
+      system "make", "update", "PYTHON_ENV=#{buildpath}/venv"
       (libexec/"bin").install "auditbeat"
       libexec.install "_meta/kibana"
 
```

Reply:
I do prefer the resource block though since it avoids a dependency on the virtualenv DSL.

Reply:
Yet using the virtualenv DSL is clearer and less manual?

Reply:
Now you know how much I don't like it.

Reply:
Okay, I pushed a new commit.